stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/ui/common/helper/ImageHeadersProvider.kt

```diff
@@ -20,9 +20,16 @@ package io.getstream.chat.android.ui.common.helper
  * Provides HTTP headers for image loading requests.
  */
 public interface ImageHeadersProvider {
-    public fun getImageRequestHeaders(): Map<String, String>
+
+    /**
+     * Returns a map of headers to be used for the image loading request.
+     *
+     * @param url The URL of the image to load.
+     * @return A map of headers to be used for the image loading request.
+     */
+    public fun getImageRequestHeaders(url: String): Map<String, String>
 }
 
 internal object DefaultImageHeadersProvider : ImageHeadersProvider {
-    override fun getImageRequestHeaders(): Map<String, String> = emptyMap()
+    override fun getImageRequestHeaders(url: String): Map<String, String> = emptyMap<String, String>()
```

Comment:
Wondering if it was a request :)

I assume this was added to determine which headers to include based on `url`, right?

Reply:
It wasn't requested, but if we had it, the customer would be able to decide which headers to add to every case.
Apart from that, it was a security bug, because this provider is used to load ALL pictures on the message list, the headers could be obtained easily by a final user.